pkgs/development/compilers/gcc/patches/default.nix

```diff
@@ -191,19 +194,30 @@ in
   sha256 = "sha256-XtykrPd5h/tsnjY1wGjzSOJ+AyyNLsfnjuOZ5Ryq9vA=";
 })
 
+# Fix undefined symbol errors when building older versions with clang
+++ optional (!atLeast11 && stdenv.cc.isClang) ./clang-genconditions.patch
```

Comment:
```suggestion
++ optional (!atLeast11 && stdenv.cc.isClang && stdenv.hostPlatform.isDarwin) ./clang-genconditions.patch
```

Reply:
This technically isn’t a Darwin-specific problem even if it’s primarily affecting x86_64-darwin. There’s an issue in GCC’s Bugzilla regarding the issue. It’s mentioned already in the commit message and at the top of the patch, but I could also include it in a comment here for clarity.

Reply:
> This technically isn’t a Darwin-specific problem even if it’s primarily affecting x86_64-darwin.

Okay but only Darwin needs this super-urgently.

Please consider committing my suggestion, and sending another PR to undo it after the branch-off.

Reply:
> It’s almost 2:30 AM here, so I need to get some sleep. 

In light of this, I'm going to commit my minor suggestion above and merge the result (you've addressed all of my concerns and this is now based on `master`).

It should be merged back to `staging-next` by the time you wake up.

Please don't hesitate to open a PR which undoes my suggested change; as long as somebody who knows using-clang-on-linux (I don't, @qyliss or @Ericson2314 probably do) signs off on it it's fine with me.